CMakeLists.txt

```diff
@@ -130,14 +130,37 @@ if(WITH_GPU)
 endif(WITH_GPU)
 
 add_subdirectory(proto)
-add_subdirectory(paddle)
-add_subdirectory(python)
 
+# "add_subdirectory(paddle)" and "add_subdirectory(python)" should be
+# placed after this block, because they depends on it.
 if(WITH_GOLANG)
```

Comment:
we can put this configuration into `configure.cmake` to keep the root CMAKE FILE as simple as possible.